Python/sysmodule.c

```diff
@@ -1799,6 +1799,49 @@ sys_getallocatedblocks_impl(PyObject *module)
 }
 
 
+/*[clinic input]
+sys._getfunc
+
+    depth: int = 1
+    /
+
+Return a function object from the call stack.
+
+If optional integer depth is 1 or more, return the function object that many
+calls below the top of the stack.  If that is deeper than the call
+stack, ValueError is raised.  If depth is 0, return the current function
+object.
+
+This is similar to sys._getframe() but cheaper because it does not
```

Comment:
Probably better to explain `sys._getframe()` in terms of `sys._getfunc`. `sys._getfunc` (or whatever it ends up being called) should be more portable.

Reply:
Added a note to the docstring and RST docs for `_getframe` instead.